xhr.bs

```diff
@@ -1021,6 +1022,13 @@ method must run these steps:
   <p>Otherwise, if the <a>synchronous flag</a> is set, run these substeps:
 
   <ol>
+   <li>
+    <p>If <a>context object</a>'s <a>relevant settings object</a> has a
+    <a>responsible document</a> and it is <em>not</em>
+    <a>allowed to use</a> the <a>Synchronous XMLHttpRequest</a>
+    feature, then run <a>handle response end-of-body</a> for a <a>network
+    error</a> and return.
```

Comment:
It seems nicer to throw during `open()`, no?

Reply:
That would be surprising to developers, I think -- there aren't currently any cases of a NetworkError being thrown during `open()`, since the fetch algorithm isn't called until send(). Since this policy can be imposed on pages which were written before this proposal, it seems that using an existing failure pattern is less likely to result in unforseen behaviour on the page.

I had originally proposed throwing an `InvalidAccessError` during `open()`, to align with the note at https://xhr.spec.whatwg.org/#sync-warning, but the rough consensus at TPAC (I wasn't actually present, this is second-hand) was around emulating a network-layer-error instead.

Reply:
Well yeah, I wouldn't want to throw a `NetworkError` there. I wonder what rationale was given in that discussion. Who would we ask?

Reply:
Note that `InvalidAccessError` would also be consistent with step 10 of `open()` today.

Reply:
I'll ask on #178, and see if we can find out who was there, and if this accurately reflects the consensus.

Reply:
https://chromium.googlesource.com/chromium/src/+/40126a62123c1e8704b2f92a6ef54eb3e6ce52db/third_party/WebKit/Source/core/xmlhttprequest/XMLHttpRequest.cpp#695 now throws InvalidAccessError in `open()`, which I think makes sense, has this been resolved but the comment thread just left open?

Reply:
https://chromium-review.googlesource.com/c/chromium/src/+/804057 changes it from `InvalidAccessError` to `NetworkError`, and causes it to occur on send(), like other network conditions would.

Reply:
I see, thanks!